### PR TITLE
HWP-404: Backend - creatorName, memberCount, and latestActivity to community

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description

This PR includes the following proposed change(s):

- Adds creatorName (Full name of community creator), memberCount, and latestActivity (date and time of latest post or comment) to community schema.
- memberCount updates when a user leaves/joins a community.
- creatorName is either user's firstName and lastName OR just firstName OR username.  
- Changed rate limit from 100 requests per minute -> 100 reqs per 15 seconds.
- Changed `creator` field to use `id` instead of `username` (the way it should have been originally).

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [x] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

Remove data from `mongo/data` directory.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested in swagger and alongside frontend changes. Joining/leaving communities, creating posts to update latestActivity.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [x] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/routes/v1/community/community.js [hunk 1]
```diff
@@ -90,15 +90,30 @@ router.post("/", async (req, res) => {
     // TODO: Validate formatting for tags in request body
     // TODO: Prevent spaces or special characters in community title
 
+    req.log.addAction("Getting creator name.");
+    const { firstName, lastName } = documents.user;
+    let creatorName;
+
+    if (firstName && firstName !== "") {
```
Comment (akroon3r) [inline app/server/routes/v1/community/community.js:97]: shouldn't firstName !== "" be enough?

File: app/server/routes/v1/post/post.js [hunk 2]
```diff
@@ -134,6 +135,13 @@ router.post("/", async (req, res) => {
     );
     req.log.addAction("Community engagement updated.");
 
+    req.log.addAction("Updating community latest activity.");
+    await Community.updateOne(
+      { title: post.community },
+      { $set: { latestActivity: moment().format("MMMM Do YYYY, h:mm:ss a") } }
+    );
+    req.log.addAction("Community latest activity updated.");
```
Comment (akroon3r) [inline app/server/routes/v1/post/post.js:143]: do we need to include the community name and timestamp here?

Comment (BradyMitch) [reply app/server/routes/v1/post/post.js:143]: Do you mean in the log? Technically the timestamp of the request will be logged, and the community title is also logged as it is part of the request body. Do you think it should be included in the log action like `"Community (Welcome) latest activity updated."`?

Comment (akroon3r) [reply app/server/routes/v1/post/post.js:143]: I think more detail would be nifty imo.

File: app/server/routes/v1/post/post.js [hunk 3]
```diff
@@ -134,6 +135,13 @@ router.post("/", async (req, res) => {
     );
     req.log.addAction("Community engagement updated.");
 
+    req.log.addAction("Updating community latest activity.");
```
Comment (BradyMitch) [inline app/server/routes/v1/post/post.js:138]:
```suggestion
    req.log.addAction(`Updating community (${req.body.community}) latest activity.`);
```

File: app/server/routes/v1/post/post.js (hunk 2 above)
Comment (BradyMitch) [inline app/server/routes/v1/post/post.js:143]:
```suggestion
    req.log.addAction(`Community (${req.body.community}) latest activity updated.`);
```

**Pull request merged**